src/app/payment/Payment.tsx

```diff
@@ -281,6 +282,7 @@ class Payment extends Component<PaymentProps & WithCheckoutPaymentProps & WithLa
         if (!isSubmittingOrder ||
             !selectedMethod ||
             selectedMethod.type === PaymentMethodProviderType.Hosted ||
+            selectedMethod.type === PaymentMethodProviderType.PPSDK ||
```

Comment:
So all of these payment methods are hosted payment methods. The only reason why they are listed is because they are incorrectly returned as "credit card" methods instead, due to the way they are implemented by the integration team. As far as I can tell, there's no real reason why they can't return the correct type, except that it involves a bit of work.

Anyway, what you have is fine I suppose... but I believe not all PPSDK methods require a redirect. So having this line means the alert won't show for credit card PPSDK methods. But you can't really tell if a PPSDK method requires a redirect because that information comes from the "steps". For example, regular CC should prompt an alert but if it triggers a 3DS challenge it shouldn't.

Anyway, I think for now this okay, at least you can test the flow. But as you said, we should put some thoughts into this - perhaps moving the `beforeunload` handler into the SDK, which can make more granular decisions.

Reply:
Yep, I know what you mean about the non-redirecting PPSDK methods - I figured we'd live without the potentially useful unload warning until a proper solution surfaces.

We could turn them all into `PaymentMethodProviderType.Hosted` (and find a different attribute to use to pivot between PPSDK/non-PPSDK), but I think centralising 'navigation' concerns might be better.

That if the SDK does the `beforeunload` _and_ the redirecting, it can more elegantly skip the `beforeunload` as needed.